R/comorbidity_index.R

```diff
@@ -154,13 +164,36 @@ diagnoses_at_admission <- function(ipdiag, erdiag) {
 #' @export
 #'
 comorbidity_index <- function(ipdiag, erdiag, map, weights, at_admission = TRUE, raw_comorbidities = FALSE) {
+  ### users can set erdiag to NULL to exclude ER diagnoses; but warning will be shown
+  if (is.null(erdiag)) {
+    cat("\n*** Based on the input you provided, only in-patient diagnoses (ipdiag) will be included in the comorbidity index.
+    If you want to include ER diagnoses, please provide the correspondig table as an input to `erdiag`. ***\n")
+  }
+
+  ### check that ipdiag/erdiag contains genc_id & diagnosis_code
+  check_input(ipdiag, c("data.table", "data.frame"),
+              colnames = c("genc_id", "diagnosis_code"),
+              coltypes = c("", "character"))
+
+  if (!is.null(erdiag)){
+    check_input(erdiag, c("data.table", "data.frame"),
+                colnames = c("genc_id", "er_diagnosis_code"),
+                coltypes = c("", "character"))
+  }
+
+
   ipdiag <- coerce_to_datatable(ipdiag)
-  erdiag <- coerce_to_datatable(erdiag)
 
-  all_diagnoses <- rbind(
-    erdiag[, .(genc_id, "diagnosis_code" = er_diagnosis_code, "diagnosis_type" = er_diagnosis_type)],
-    ipdiag[, .(genc_id, diagnosis_code, diagnosis_type)]
-  )
+  if (!is.null(erdiag)){
+    erdiag <- coerce_to_datatable(erdiag)
+
+    all_diagnoses <- rbind(
+      erdiag[, .(genc_id, "diagnosis_code" = er_diagnosis_code, "diagnosis_type" = er_diagnosis_type)],
+      ipdiag[, .(genc_id, diagnosis_code, diagnosis_type)]
+    )
+  } else {
+    all_diagnoses <- ipdiag[, .(genc_id, diagnosis_code, diagnosis_type)]
+  }
```

Comment:
I have a similar style comment here as above. But please feel free to take or leave based on your judgement. It has no real consequence.

Reply:
Great, point! I agree this is much nicer to read. I've accepted the change you suggested above and also made a similar change here.